src/RosterModel.cpp

```diff
@@ -193,3 +193,20 @@ void RosterModel::setUnreadMessageCountOfJid(const QString* jid_, const int coun
 
 	submit();
 }
+
+void RosterModel::setLastMessageForJid(QString *jid, QString *message)
+{
+	QSqlQuery query;
+	query.prepare(QString("UPDATE Roster SET lastMessage = %1 WHERE jid = '%2'")
```

Comment:
The `lastMessage` row in the SQL table already exists and was just unused before.